docs/developer/architecture/core/saved-objects-service.asciidoc

```diff
@@ -157,242 +161,1093 @@ Saved Object types should never use `dynamic: true` as this can cause an
+
+The schemas associated with this version. Schemas are used to validate
+or convert SO documents at various stages of their lifecycle.
+
+The currently available schemas are:
+
+===== forwardCompatibility
+
+This is a new concept introduced by model versions. This schema is used
+for inter-version compatibility.
+
+When retrieving a savedObject document from an index, if the version of
+the document is higher than the latest version known of the Kibana
+instance, the document will go through the `forwardCompatibility` schema
+of the associated model version.
+
+*Important:* These conversion mechanism shouldn’t assert the data
+itself, and only strip unknown fields to convert the document to the
+*shape* of the document at the given version.
+
+Basically, this schema should keep all the known fields of a given
+version, and remove all the unknown fields, without throwing.
+
+Forward compatibility schema can be implemented in two different ways.
+
+[arabic]
+. Using `config-schema`
+
+_Example of schema for a version having two fields: someField and
+anotherField_
+
+[source,ts]
+----
+const versionSchema = schema.object(
+  {
+    someField: schema.maybe(schema.string()),
+    anotherField: schema.maybe(schema.string()),
+  },
+  { unknowns: 'ignore' }
+);
+----
+
+*Important:* Note the `{ unknowns: 'ignore' }` in the schema’s options.
+This is required when using `config-schema` based schemas, as this what
+will evict the additional fields without throwing an error.
+
+[arabic, start=2]
+. Using a plain javascript function
+
+_Example of schema for a version having two fields: someField and
+anotherField_
+
+[source,ts]
+----
+const versionSchema: SavedObjectModelVersionEvictionFn = (attributes) => {
+  const knownFields = ['someField', 'anotherField'];
+  return pick(attributes, knownFields);
 }
 ----
 
-===== Hidden from the HTTP APIs 
+*note:* _Even if highly recommended, implementing this schema is not
+strictly required. Type owners can manage unknown fields and
+inter-version compatibility themselves in their service layer instead._
 
-When a saved object is registered as hidden from the HTTP APIs, it will remain exposed to the global Saved Objects client:
+===== create
 
-[source,typescript]
+This is a direct replacement for
+https://github.com/elastic/kibana/blob/9b330e493216e8dde3166451e4714966f63f5ab7/packages/core/saved-objects/core-saved-objects-server/src/saved_objects_type.ts#L75-L82[the
+old SavedObjectType.schemas] definition, now directly included in the
+model version definition.
+
+As a refresher the `create` schema is a `@kbn/config-schema` object-type
+schema, and is used to validate the properties of the document during
+`create` and `bulkCreate` operations.
+
+*note:* _Implementing this schema is optional, but still recommended, as
+otherwise there will be no validating when importing objects_
+
+For implementation examples, refer to <<saved-objects-service-use-case-examples, Use-case examples>>.
+=== Use-case examples
```

Comment:
I think if you want to link to this section, you need an anchor, for example:
```suggestion

[[saved-objects-service-use-case-examples]]
=== Use-case examples
```

Reply:
oops,
Thanks!